pkg/util/restoptions/helpers.go

```diff
@@ -0,0 +1,93 @@
+package restoptions
+
+import (
+	"strings"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/genericapiserver"
+	"k8s.io/kubernetes/pkg/registry/cachesize"
+	"k8s.io/kubernetes/pkg/registry/generic/registry"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+)
+
+// defaultKeyFunctions sets the default behavior for storage key generation onto a Store.
+func defaultKeyFunctions(store *registry.Store, prefix string) {
```

Comment:
this is what we call "completion" in the genericapiserver. Suggestion: rename to `completeKeyFunctions`